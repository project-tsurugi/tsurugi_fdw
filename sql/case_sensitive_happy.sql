/* Test case: happy path - Case sensitivity */
-- Test setup: DDL of the Tsurugi
SELECT tg_execute_ddl('
  CREATE TABLE fdw_case_table (
    col INT,
    Col INT
  )
', 'tsurugidb');

-- Test setup: DDL of the PostgreSQL
CREATE FOREIGN TABLE fdw_case_table (
  col integer,
  "Col" integer
) SERVER tsurugidb;

-- Test
/* Fix tsurugi-issues#568 */
INSERT INTO fdw_case_table VALUES (1000, 1);
INSERT INTO public.fdw_case_table (col, "Col") VALUES (2000, 2);
INSERT INTO "public".fdw_case_table (col) VALUES (3000);
INSERT INTO "public"."fdw_case_table" (Col) VALUES (4);
INSERT INTO Public.fdw_case_table ("col") VALUES (5000);
INSERT INTO PUBLIC.fdw_case_table ("Col") VALUES (6);
SELECT * FROM public.fdw_case_table ORDER BY col, Col DESC;

UPDATE fdw_case_table SET col = col + 100;
UPDATE public.fdw_case_table SET Col = Col + 10;
UPDATE "public".fdw_case_table SET "col" = "col" + 100 WHERE col > 5000;
UPDATE public."fdw_case_table" SET "Col" = "Col" + 100 WHERE Col > 12;
UPDATE "public"."fdw_case_table" SET "col" = col + 100 WHERE "col" > 3000;
UPDATE Public.fdw_case_table SET "col" = Col * 10 WHERE col IS NULL;
UPDATE PUBLIC.fdw_case_table SET "Col" = col / 100 WHERE "Col" IS NULL;
SELECT * FROM fdw_case_table ORDER BY col;
SELECT * FROM Public.fdw_case_table ORDER BY Col;

DELETE FROM fdw_case_table WHERE col = 1100;
DELETE FROM public.fdw_case_table WHERE "col" > 5000;
DELETE FROM "public".fdw_case_table WHERE Col > 114;
DELETE FROM public."fdw_case_table" WHERE "col" >= 3200;
DELETE FROM "public"."fdw_case_table" WHERE "Col" > 100;
DELETE FROM Public.fdw_case_table WHERE col <> 2100;
DELETE FROM PUBLIC.fdw_case_table WHERE Col <> 12;
SELECT * FROM Public.fdw_case_table;

-- Test teardown: DDL of the PostgreSQL
DROP FOREIGN TABLE fdw_case_table;
-- Test teardown: DDL of the Tsurugi
SELECT tg_execute_ddl('DROP TABLE fdw_case_table', 'tsurugidb');
