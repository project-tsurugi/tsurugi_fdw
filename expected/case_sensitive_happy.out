/* Test case: happy path - Case sensitivity */
-- Test setup: DDL of the Tsurugi
SELECT tg_execute_ddl('
  CREATE TABLE fdw_case_table (
    col INT,
    Col INT
  )
', 'tsurugidb');
 tg_execute_ddl 
----------------
 CREATE TABLE
(1 row)

-- Test setup: DDL of the PostgreSQL
CREATE FOREIGN TABLE fdw_case_table (
  col integer,
  "Col" integer
) SERVER tsurugidb;
-- Test
/* Fix tsurugi-issues#568 */
INSERT INTO fdw_case_table VALUES (1000, 1);
INSERT INTO public.fdw_case_table (col, "Col") VALUES (2000, 2);
INSERT INTO "public".fdw_case_table (col) VALUES (3000);
INSERT INTO "public"."fdw_case_table" (Col) VALUES (4);
INSERT INTO Public.fdw_case_table ("col") VALUES (5000);
INSERT INTO PUBLIC.fdw_case_table ("Col") VALUES (6);
SELECT * FROM public.fdw_case_table ORDER BY col, Col DESC;
 col  | Col 
------+-----
      |   6
      |   4
 1000 |   1
 2000 |   2
 3000 |    
 5000 |    
(6 rows)

UPDATE fdw_case_table SET col = col + 100;
UPDATE public.fdw_case_table SET Col = Col + 10;
UPDATE "public".fdw_case_table SET "col" = "col" + 100 WHERE col > 5000;
UPDATE public."fdw_case_table" SET "Col" = "Col" + 100 WHERE Col > 12;
UPDATE "public"."fdw_case_table" SET "col" = col + 100 WHERE "col" > 3000;
UPDATE Public.fdw_case_table SET "col" = Col * 10 WHERE col IS NULL;
UPDATE PUBLIC.fdw_case_table SET "Col" = col / 100 WHERE "Col" IS NULL;
SELECT * FROM fdw_case_table ORDER BY col;
 col  | Col 
------+-----
 1100 |  11
 1140 | 114
 1160 | 116
 2100 |  12
 3200 |  32
 5300 |  53
(6 rows)

SELECT * FROM Public.fdw_case_table ORDER BY Col;
 col  | Col 
------+-----
 1100 |  11
 2100 |  12
 3200 |  32
 5300 |  53
 1140 | 114
 1160 | 116
(6 rows)

DELETE FROM fdw_case_table WHERE col = 1100;
DELETE FROM public.fdw_case_table WHERE "col" > 5000;
DELETE FROM "public".fdw_case_table WHERE Col > 114;
DELETE FROM public."fdw_case_table" WHERE "col" >= 3200;
DELETE FROM "public"."fdw_case_table" WHERE "Col" > 100;
DELETE FROM Public.fdw_case_table WHERE col <> 2100;
DELETE FROM PUBLIC.fdw_case_table WHERE Col <> 12;
SELECT * FROM Public.fdw_case_table;
 col  | Col 
------+-----
 2100 |  12
(1 row)

-- Test teardown: DDL of the PostgreSQL
DROP FOREIGN TABLE fdw_case_table;
-- Test teardown: DDL of the Tsurugi
SELECT tg_execute_ddl('DROP TABLE fdw_case_table', 'tsurugidb');
 tg_execute_ddl 
----------------
 DROP TABLE
(1 row)

